### PR TITLE
DDF-3298 Made string values in PersistentStore into an enum

### PR DESCRIPTION
**Pull request by josephthweatt**
#### What does this PR do?
Resolves a vulnerability found by SonarQube

#### Who is reviewing it? 
@emmberk  @mcalcote 
#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Choose 2 committers to review/merge the PR. 
@clockard
@coyotesqrl 
#### What are the relevant tickets?
[DDF-3298](https://codice.atlassian.net/browse/)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by josephthweatt** [comment]
build now

File: platform/persistence/platform-persistence-core-api/src/main/java/org/codice/ddf/persistence/PersistentStore.java [hunk 1]
```diff
@@ -13,43 +13,50 @@
...
-    String NOTIFICATION_TYPE = "notification";
-    String ACTIVITY_TYPE = "activity";
-    String WORKSPACE_TYPE = "workspace";
-    String PREFERENCES_TYPE = "preferences";
-    String USER_ATTRIBUTE_TYPE = "attributes";
-    String SUBSCRIPTION_TYPE = "subscriptions";
-    String EVENT_SUBSCRIPTIONS_TYPE = "event_subscriptions";
-    String ALERT_TYPE = "alerts";
-    String DECANTER_TYPE = "decanter";
+    enum PERSISTENCE_TYPES {
```
Comment (clockard) [inline platform/persistence/platform-persistence-core-api/src/main/java/org/codice/ddf/persistence/PersistentStore.java]: I think this should maybe be renamed PersistenceType

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by josephthweatt** [comment]
build now

**Comment by vinamartin** [comment]
build now

**Comment by josephthweatt** [comment]
Successfully built downstream projects off of these changes.

**Pull request merged**